### PR TITLE
Target selfcal updates

### PR DESCRIPTION
**Pull request by jwpetley**
I think this is the correct format for target selfcal processing with the current facetselfcal setup. 

I changed the `run_selfcal.py` script to copy any config file to the destination directory with the correct name which I think is easier. I have added a `targetselfcal_config.txt` file with the auto settings.

Finally, the phaseup step has been improved to do the correct averaging. I believe this is all valid with the corrected data column?

### CODE REVIEW

**Comment by lmorabit** [comment]
this looks good except that the delay calibration solutions need to be applied with the facetselfcal.py with the `preapplyH5-list` parameter; this should be the same h5 that's currently applied in the apply_delay step in Delay-Calibration.  I think we'll need a version of the `createmap_selfcal` step and possibly the copy step after, but it doesn't need to be expanded.

**Comment by jwpetley** [comment]
Are we removing the apply_delay step from the `Delay-Calibration.parset` ? Or should we be operating on `DATA` instead of `CORRECTED_DATA` in `Split-Directions.parset`. 

I'll work on the `createmap_selfcal` step

**Comment by lmorabit** [comment]
Yes, we are going to remove the apply_delay step, because that is doing things out of order -- applying solutions before the beam, but the beam should only be applied after phase-shifting.  It probably won't make a super big difference but we want to do it correctly.  We can probably remove most of the steps after delay_solve, although we may want to keep the final make summary?

Split-Directions should then use either DATA or DATA_DI_CORRECTED, the same as the {{ datacol }} used in the phaseup in delay-calibration.

**Comment by jwpetley** [comment]
Ok have implemented that now. Definitely check that I've done the mapfile stuff properly though

File: Split-Directions.parset [hunk 1]
```diff
@@ -120,19 +125,15 @@ dppp_phaseup.argument.msin.datacolumn           = {{ data_col }}
 dppp_phaseup.argument.msout.datacolumn          = DATA
```
Comment (lmorabit) [inline Split-Directions.parset]: change to {{ data_ col }}

File: Split-Directions.parset [hunk 2]
```diff
@@ -54,7 +59,7 @@
 ##########################################################
 
 ## data column from Delay-Calibration
-! data_col = CORRECTED_DATA 
+! data_col = DATA
```
Comment (lmorabit) [inline Split-Directions.parset]: change to

! data_col = DATA   ## set to DATA_DI_CORRECTED if you applied ddf solutions in Delay-Calibration

File: Split-Directions.parset [hunk 3]
```diff
@@ -182,21 +183,26 @@ phaseup_concat.argument.msout.overwrite        = True
 phaseup_concat.argument.steps                  = [count]
 phaseup_concat.argument.numthreads             = {{ max_dppp_threads }}
 
+# generate mapfile of the output file
+createmap_selfcal.control.kind             = plugin
+createmap_selfcal.control.type             = createMapfile
+createmap_selfcal.control.method           = mapfile_from_folder
+createmap_selfcal.control.mapfile_dir      = input.output.mapfile_dir
+createmap_selfcal.control.filename         = createmap_selfcal.mapfile
+createmap_selfcal.control.folder           = {{ job_directory }}/delay_solve
```
Comment (lmorabit) [inline Split-Directions.parset]: this should be {{ job_directory }}/../Delay-Calibration/delay_solve 

but check Delay-Calibration for how it's done

**Review by lmorabit — COMMENTED** [review]
i've left some individual comments

**Comment by cyriltasse** [comment]
Hi - I'm trying this and I have

```
2023-04-21 06:57:23 ERROR   genericpipeline.executable_args: Remote process python /opt/lofar/lofar/lib64/python2.7/site-packages/lofarpipe/recipes/nodes/executable_args.py ['/data/levangelista/LOFAR_VLBI_PIPELINE/RUN_DIR/Split-Directions/../Delay-Calibration/L402151_SB244_uv.dppp_1273342BDt_123MHz.msdpppconcat', '/opt/lofar/lofar/bin/NDPPP', [], {'msin.datacolumn': 'DATA', 'shift.type': 'phaseshift', 'applybeam.beammode': 'full', 'average1.freqresolution': '48.82kHz', 'average2.timeresolution': '32.', 'average1.type': 'averager', 'applybeam.type': 'applybeam', 'average2.freqresolution': '390.56kHz', 'shift.phasecenter': '["181.967050311481deg","38.9331038926687deg"]', 'steps': '[shift, average1, applybeam, average2]', 'msout.overwrite': 'True', 'applybeam.direction': '["181.967050311481deg","38.9331038926687deg"]', 'average2.type': 'averager', 'msin': '/data/levangelista/LOFAR_VLBI_PIPELINE/RUN_DIR/Split-Directions/../Delay-Calibration/L402151_SB244_uv.dppp_1273342BDt_123MHz.msdpppconcat', 'msout.storagemanager': 'dysco', 'numthreads': '4', 'msout.datacolumn': '{{ data_col}}', 'average1.timeresolution': '4.', 'msout': '/data/levangelista/LOFAR_VLBI_PIPELINE/RUN_DIR/Split-Directions/../Delay-Calibration/ILTJ120752.09+385559.1_L402151_SB244_uv.dppp_1273342BDt_123MHz.msdpppconcat'}, '/data/levangelista/LOFAR_VLBI_PIPELINE/RUN_DIR/Split-Directions', False, {'args_format_option_argument': '=', 'args_format_option': '-', 'args_format': 'lofar', 'args_formatlongoption': '--', 'args_format_argument': ''}, {'LOFARDATAROOT': '/opt/lofar/data', 'USER_PATH': '/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/cep/lofar/bin:/cep/lofar/scripts:/opt/dell/srvadmin/bin:/home/levangelista/.local/bin:/home/levangelista/bin:/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin', 'PYTHONPATH': '/opt/lofar:/opt/lofar/ddf-pipeline/scripts:/opt/lofar/ddf-pipeline/utils:/opt/lofar/DDFacet/lib/python2.7/site-packages:/opt/lofar/killMS:/opt/lofar/killMS/Predict:/opt/lofar/killMS/Array:/opt/lofar/killMS/Array/Dot:/opt/lofar/killMS/Gridder:/opt/lofar/DDFacet/bin:/opt/lofar/DynSpecMS:/opt/lofar/DPPP/lib/python3.7/site-packages:/opt/lofar/DPPP/usermodules:/opt/lofar/lofar/lib64/python2.7/site-packages', 'OMP_NUM_THREADS': '8', 'LOFARROOT': '/opt/lofar/lofar', 'PATH': '/opt/lofar/pyenv-py2/bin:/opt/lofar/DDFacet/bin:/opt/lofar/DDFacet/src/SkyModel:/opt/lofar/DynSpecMS/:/opt/lofar/killMS:/opt/lofar/ddf-pipeline/scripts:/usr/lib64/openmpi/bin:/opt/lofar/wsclean/bin:/opt/lofar/lofar/bin:/opt/hdf5/bin:/opt/lofar/dysco/bin:/opt/lofar/difmap/uvf_difmap:/opt/lofar/DPPP/bin:/opt/lofar/ds9/bin:/opt/lofar/casacore/bin:/opt/lofar/aoflagger/bin:/opt/montage/Montage-6.0/bin:/usr/local/cuda/bin:/opt/hdf5/bin:/opt/lofar/lofar/sbin:/opt/lofar/lofar/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'LD_LIBRARY_PATH': '/opt/lofar/aoflagger/lib:/opt/lofar/casacore/lib:/opt/lofar/DPPP/lib:/opt/lofar/dysco/lib:/opt/lofar/EveryBeam/lib:/opt/hdf5/lib:/opt/lofar/idg/lib:/opt/lofar/lofar/lib:/opt/lofar/lofar/lib64:/opt/lofar/LOFARBeam/lib:/opt/lofar/lofarstman/lib64:/usr/local/cuda/lib64:/opt/hdf5/lib:/opt/intel/mkl/lib/intel64/:/usr/lib64/openmpi/lib/:/opt/lofar/lofar/lib64:/.singularity.d/libs'}] failed on localhost
2023-04-21 06:57:24 WARNING genericpipeline.executable_args: /opt/lofar/lofar/lib64/python2.7/site-packages/lofarpipe/support/utilities.pyc : Using default subprocess module!
2023-04-21 06:57:24 WARNING genericpipeline.executable_args: Debug: registered context Global=0
2023-04-21 06:57:24 ERROR   node.nancep7.obs-nancay.fr.executable_args: Command '/opt/lofar/lofar/bin/NDPPP' returned non-zero exit status 1
2023-04-21 06:57:24 WARNING node.nancep7.obs-nancay.fr.executable_args.L402151_SB244_uv.dppp_1273342BDt_134MHz.msdpppconcat: /opt/lofar/lofar/bin/NDPPP stderr: 
std exception detected: Currently only the DATA column can be used as output when writing a new MS
```

I guess this line https://github.com/lmorabit/lofar-vlbi/blob/0414cf5c3fcd34302bc685181beae21c7bf041bd/Split-Directions.parset#L125 might be changed to `dppp_phaseup.argument.msout.datacolumn          = DATA`?

**Comment by lmorabit** [comment]
@cyriltasse yes, you're correct -- I've made that change now

**Comment by cyriltasse** [comment]
Thanks!

**Pull request merged**